pkg/accounting/accounting.go

```diff
@@ -226,6 +231,66 @@ func (a *Accounting) getPeerBalance(peer swarm.Address) (*PeerBalance, error) {
 	return peerBalance, nil
 }
 
+// Balances gets balances for all peers, first from memory, than completing from store
+func (a *Accounting) Balances() (map[string]int64, error) {
+	peersBalances := make(map[string]int64)
+
+	a.balancesMu.Lock()
+	for peer, balance := range a.balances {
+		peersBalances[peer] = balance.balance
+	}
+	a.balancesMu.Unlock()
+
+	err := a.completeFromStore(peersBalances)
+	if err != nil {
+		return nil, err
+	}
+
+	return peersBalances, nil
+}
+
+// Get balances from store for keys (peers) that do not already exist in argument map
+// Used to get all balances not loaded in memory at the time the Balances() function is called
+func (a *Accounting) completeFromStore(s map[string]int64) error {
+	err := a.store.Iterate(balancesPrefix, func(key, val []byte) (stop bool, err error) {
+		addr, err := balanceKeyPeer(key)
+		if err != nil {
+			a.logger.Debugf("could not parse address from key: %v: %v", string(key), err)
+			return false, nil
+		}
+		if _, ok := s[addr.String()]; !ok {
+			var storevalue int64
+			err = a.store.Get(peerBalanceKey(addr), &storevalue)
+			if err != nil {
+				a.logger.Debugf("store operation error for peer %v: %v", addr.String(), err)
+				return false, nil
+			}
+
+			s[addr.String()] = storevalue
+		}
+		return false, nil
+	})
```

Comment:
it is a bit unclear to me how we are making use of this `Iterate` call. it seems like the in-line func you are passing to it as a parameter never returns an error, so unless i'm seeing it wrong, `completeFromStore` will _always_ return `nil`. 

please correct me if i'm mistaken, and if not, could you please explain why we are silencing errors from the store iteration?